api/services/contact_fields_mapping_service.py

```diff
@@ -0,0 +1,107 @@
+            if not user_details:
+                return False
+
+            contact_groups = jsonData["contact"]["groups"]
+
+            for contact_group in contact_groups:
+                contact_group_name = contact_group["name"]
+                user_contact_group_mapping = (
+                    models.ContactFieldsMapping.query.get_by_group_name(
+                        contact_group_name
+                    )
+                )
+                if user_contact_group_mapping:
+                    self.process_contact_groups_data(
+                        user_contact_group_mapping, user_details
+                    )
+        except Exception as e:
+            print(f"Error occurred while handling custom field mapping. Exception: {e}")
+
+    def process_contact_fields_data(self, user_contact_field_details, user_details):
+        try:
+            for user_contact_field_detail in user_contact_field_details:
+                table_object = helpers.get_class_by_tablename(
+                    user_contact_field_detail.mapped_table_name
+                )
+                if table_object:
+                    column_name = user_contact_field_detail.mapped_table_column_name
+                    mapped_table_column_value = (
+                        user_contact_field_detail.mapped_table_column_value
+                    )
+                    self.update_mapped_fields(
+                        table_object,
+                        column_name,
+                        mapped_table_column_value,
+                        user_details,
+                    )
+        except Exception as e:
+            print(f"Exception occurred: {e}")
+
+    def process_contact_groups_data(self, user_contact_group_details, user_details):
+        try:
+            for user_contact_group_detail in user_contact_group_details:
+                table_object = helpers.get_class_by_tablename(
+                    user_contact_group_detail.mapped_table_name
+                )
+                if table_object:
+                    column_name = user_contact_group_detail.mapped_table_column_name
+                    mapped_table_column_value = (
+                        user_contact_group_detail.mapped_table_column_value
+                    )
+                    self.update_mapped_fields(
+                        table_object,
+                        column_name,
+                        mapped_table_column_value,
+                        user_details,
+                    )
+        except Exception as e:
+            print(f"Exception occurred: {e}")
+
+    def update_mapped_fields(
+        self, table_object, column_name, mapped_table_column_value, user_details
+    ):
+        try:
+            table_data = table_object.get_by_user_id(user_details.id)
+            if table_data:
+                setattr(table_data, column_name, mapped_table_column_value)
+                db.session.commit()
```

Comment:
Let's add a check here that will only update the data if the existing value is different than the new value. This way we can reduce the database call to update statements as while updating the table gets locked and no other update process happens which can cause a huge delay if we update the data for all cases.